staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -570,6 +573,39 @@ func GetFieldManagerFlag(cmd *cobra.Command) string {
 	return GetFlagString(cmd, "field-manager")
 }
 
+type ValidationDirective string
+
+const (
+	ValidationDirectiveStrict ValidationDirective = "Strict"
+	ValidationDirectiveWarn   ValidationDirective = "Warn"
+	ValidationDirectiveIgnore ValidationDirective = "Ignore"
+)
+
+func GetValidationDirective(cmd *cobra.Command) (ValidationDirective, error) {
+	var validateFlag = GetFlagString(cmd, "validate")
+	b, err := strconv.ParseBool(validateFlag)
+	if err != nil {
+		switch validateFlag {
+		case cmd.Flag("validate").NoOptDefVal:
+			return ValidationDirectiveStrict, nil
+		case "strict":
+			return ValidationDirectiveStrict, nil
+		case "warn":
+			return ValidationDirectiveWarn, nil
+		case "ignore":
+			return ValidationDirectiveIgnore, nil
+		default:
+			return ValidationDirectiveStrict, fmt.Errorf(`Invalid validate value (%v). Must be "true", "false", "strict", "warn", or "ignore".`, validateFlag)
+		}
+	}
+	// The flag was a boolean
+	if b {
+		// TODO: should we warn about deprecating the boolean value in favor of {strict, warn, ignore}?
```

Comment:
I would not expect to warn. I can't think of any benefit to making people change --validate=true to --validate=strict and --validate=false to --validate=ignore

Reply:
Yeah, it's because for dry-run we wanted to deprecate the boolean form, not sure we want to deprecate it here.